server/src/main/java/org/opensearch/search/pipeline/SearchPipelineStats.java

```diff
@@ -333,6 +392,127 @@ OperationStats getStats() {
         }
     }
 
+    static class FactoryStats implements ToXContentFragment, Writeable {
+        private final String factoryType;
+        private final OperationStats evaluationStats;
+        private final OperationStats generationStats;
+
+        FactoryStats(String factoryType, OperationStats evaluationStats, OperationStats generationStats) {
+            this.factoryType = factoryType;
+            this.evaluationStats = evaluationStats;
+            this.generationStats = generationStats;
+        }
+
+        FactoryStats(StreamInput in) throws IOException {
+            factoryType = in.readString();
+            evaluationStats = new OperationStats(in);
+            generationStats = new OperationStats(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(factoryType);
+            evaluationStats.writeTo(out);
+            generationStats.writeTo(out);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject(factoryType);
+            builder.field("type", factoryType);
+            builder.startObject("evaluation_stats");
+            evaluationStats.toXContent(builder, params);
+            builder.endObject();
+            builder.startObject("generation_stats");
+            generationStats.toXContent(builder, params);
+            builder.endObject();
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            FactoryStats that = (FactoryStats) o;
+            return factoryType.equals(that.factoryType)
+                && evaluationStats.equals(that.evaluationStats)
+                && generationStats.equals(that.generationStats);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(factoryType, evaluationStats, generationStats);
+        }
+    }
+
+    static class FactoryDetailStats implements ToXContentFragment, Writeable {
```

Comment:
Why do we need factory stats in the search pipeline stats? Just processors should be enough